metrics/stackdriver_exporter.go

```diff
@@ -140,21 +140,20 @@ func newStackdriverExporter(config *metricsConfig, logger *zap.SugaredLogger) (v
 
 // getStackdriverExporterClientOptions creates client options for the opencensus Stackdriver exporter from the given stackdriverClientConfig.
 // On error, an empty array of client options is returned.
-func getStackdriverExporterClientOptions(sdconfig *StackdriverClientConfig) ([]option.ClientOption, error) {
+func getStackdriverExporterClientOptions(config *metricsConfig) ([]option.ClientOption, error) {
 	var co []option.ClientOption
-	if sdconfig.UseSecret && useStackdriverSecretEnabled {
-		secret, err := getStackdriverSecret(sdconfig)
-		if err != nil {
-			return co, err
+
+	if config.stackdriverClientConfig.UseSecret {
```

Comment:
If `useStackdriverSecretEnabled` is false, config.secret will be nil then an error will be thrown. This is different compared to original code. Is it intended change?

Reply:
Tricky to see, but this error is actually returned and logged as just a warning in newStackdriverExporter. 

This message is really the only sign to metrics package consumers that they need to call SetStackdriverSecretLocation to enable Stackdriver secret option. I added a comment that SetStackdriverSecretLocation must be called to make it clearer to Knative developers.

The SetStackdriverSecretLocation requirement was to prevent accidental attempts to read and use the default Secret namespace/name ("default/stackdriver-service-account-key") if a user turned the flag on in config-observability, but knative/serving really wanted the Secret to be "knative-serving/stackdriver-service-account-key").